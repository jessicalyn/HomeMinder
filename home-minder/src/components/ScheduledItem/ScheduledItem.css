.scheduled-item {
  padding: 0 1rem;
  text-align: left
}

.scheduled-date {
  text-decoration: underline
}

.scheduled-name {
  margin-top: 0;
  margin-right: 1.5rem
}

.item-block {
  display: flex;
}