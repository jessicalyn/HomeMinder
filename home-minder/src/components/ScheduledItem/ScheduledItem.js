import React from 'react';
import './ScheduledItem.css';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';

const ScheduledItem = ({ room, name, userScheduled, id }) => {
  userScheduled = new Date(userScheduled)
  const scheduledString = userScheduled.toDateString()
  return <div className="scheduled-item">
    <p className="scheduled-date">{ scheduledString }</p>
    <div className="item-block">
      <p className="scheduled-name">{ room } { name }</p>
      <NavLink to={`/item/${id}`}>View Reminder</NavLink>
    </div>
  </div>
}

export default ScheduledItem

ScheduledItem.propTypes = {
  room: PropTypes.string,
  name: PropTypes.string,
  userScheduled: PropTypes.string,
  id: PropTypes.number
}