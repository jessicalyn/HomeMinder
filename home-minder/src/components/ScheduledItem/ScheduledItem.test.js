import React from 'react'
import ScheduledItem from './ScheduledItem';
import { shallow } from 'enzyme'

describe('ScheduledItem', () => {
  let wrapper
  const item = {
    id: 1,
    name: "Refrigerator Water Filter",
    room: "Kitchen",
    userScheduled: "2019-04-11T06:00:00.000Z"
  }

  it('should match the snapshot with data passed in', () => {
    wrapper = shallow(<ScheduledItem {...item} />)
    expect(wrapper).toMatchSnapshot()
  })
})