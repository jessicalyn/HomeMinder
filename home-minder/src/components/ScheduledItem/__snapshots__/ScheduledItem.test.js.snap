// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScheduledItem should match the snapshot with data passed in 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ScheduledItem
    id={1}
    name="Refrigerator Water Filter"
    room="Kitchen"
    userScheduled="2019-04-11T06:00:00.000Z"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="scheduled-date"
        >
          Thu Apr 11 2019
        </p>,
        <div
          className="item-block"
        >
          <p
            className="scheduled-name"
          >
            Kitchen
             
            Refrigerator Water Filter
          </p>
          <NavLink
            to="/item/1"
          >
            View Reminder
          </NavLink>
        </div>,
      ],
      "className": "scheduled-item",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Thu Apr 11 2019",
          "className": "scheduled-date",
        },
        "ref": null,
        "rendered": "Thu Apr 11 2019",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="scheduled-name"
            >
              Kitchen
               
              Refrigerator Water Filter
            </p>,
            <NavLink
              to="/item/1"
            >
              View Reminder
            </NavLink>,
          ],
          "className": "item-block",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Kitchen",
                " ",
                "Refrigerator Water Filter",
              ],
              "className": "scheduled-name",
            },
            "ref": null,
            "rendered": Array [
              "Kitchen",
              " ",
              "Refrigerator Water Filter",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "View Reminder",
              "to": "/item/1",
            },
            "ref": null,
            "rendered": "View Reminder",
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="scheduled-date"
          >
            Thu Apr 11 2019
          </p>,
          <div
            className="item-block"
          >
            <p
              className="scheduled-name"
            >
              Kitchen
               
              Refrigerator Water Filter
            </p>
            <NavLink
              to="/item/1"
            >
              View Reminder
            </NavLink>
          </div>,
        ],
        "className": "scheduled-item",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Thu Apr 11 2019",
            "className": "scheduled-date",
          },
          "ref": null,
          "rendered": "Thu Apr 11 2019",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="scheduled-name"
              >
                Kitchen
                 
                Refrigerator Water Filter
              </p>,
              <NavLink
                to="/item/1"
              >
                View Reminder
              </NavLink>,
            ],
            "className": "item-block",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Kitchen",
                  " ",
                  "Refrigerator Water Filter",
                ],
                "className": "scheduled-name",
              },
              "ref": null,
              "rendered": Array [
                "Kitchen",
                " ",
                "Refrigerator Water Filter",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "View Reminder",
                "to": "/item/1",
              },
              "ref": null,
              "rendered": "View Reminder",
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
